test/handlers/UpgradeRegressionManager.handler.sol

```diff
@@ -0,0 +1,264 @@
+    if (_rollbackId == 0) {
+      return;
+    }
+
+    RollbackProposal storage _rollbackProposal = proposals[_rollbackId];
+
+    // Check if rollback is eligible to queue
+    if (!urm.isRollbackEligibleToQueue(_rollbackProposal.rollbackId)) {
+      return;
+    }
+
+    vm.prank(guardian);
+    try urm.queue(
+      _rollbackProposal.targets, _rollbackProposal.values, _rollbackProposal.calldatas, _rollbackProposal.description
+    ) {
+      _rollbackProposal.state = ProposalState.QUEUED;
+    } catch {}
+  }
+
+  /// @notice Execute a valid rollback proposal
+  /// @param _randomIndex Used to randomly select an executable proposal
+  function executeRollback(uint256 _randomIndex) public {
+    uint256 _rollbackId = getRandomExecutableRollbackId(_randomIndex);
+    if (_rollbackId == 0) {
+      return;
+    }
+
+    RollbackProposal storage _rollbackProposal = proposals[_rollbackId];
+
+    // Check if rollback is ready to execute
+    if (!urm.isRollbackReadyToExecute(_rollbackProposal.rollbackId)) {
+      return;
+    }
+
+    vm.prank(guardian);
+    try urm.execute(
+      _rollbackProposal.targets, _rollbackProposal.values, _rollbackProposal.calldatas, _rollbackProposal.description
+    ) {
+      _rollbackProposal.state = ProposalState.EXECUTED;
+    } catch {}
+  }
+
+  /// @notice Cancel a valid rollback proposal
+  /// @param _randomIndex Used to randomly select a cancellable proposal
+  function cancelRollback(uint256 _randomIndex) public {
+    uint256 _rollbackId = getRandomQueueableRollbackId(_randomIndex);
+    if (_rollbackId == 0) {
+      return;
+    }
+
+    RollbackProposal storage _rollbackProposal = proposals[_rollbackId];
+
+    vm.prank(admin);
+    try urm.cancel(
+      _rollbackProposal.targets, _rollbackProposal.values, _rollbackProposal.calldatas, _rollbackProposal.description
+    ) {
+      _rollbackProposal.state = ProposalState.CANCELLED;
+    } catch {}
+  }
+
+  /// @notice Get a random rollback ID
+  /// @param _randomIndex Used to randomly select a proposal
+  /// @return The rollback ID of a randomly selected proposal, or 0 if none exist
+  function getRandomRollbackId(uint256 _randomIndex) public view returns (uint256) {
+    uint256 _index = bound(_randomIndex, 0, rollbackIds.length - 1);
+    return rollbackIds[_index];
+  }
+
+  /// @notice Get a random queueable rollback ID
+  /// @param _randomIndex Used to randomly select a queueable proposal
+  /// @return The rollback ID of a randomly selected queueable proposal, or 0 if none exist
+  function getRandomQueueableRollbackId(uint256 _randomIndex) public view returns (uint256) {
+    uint256[] memory _matchingIds = new uint256[](rollbackIds.length);
+    uint256 _matchCount = 0;
+
+    for (uint256 i = 0; i < rollbackIds.length; i++) {
+      if (proposals[rollbackIds[i]].state == ProposalState.PROPOSED) {
+        _matchingIds[_matchCount] = rollbackIds[i];
+        _matchCount++;
+      }
+    }
+
+    if (_matchCount == 0) {
+      return 0;
```

Comment:
As discussed on the call, it's reasonable to revert here. We want *most* calls not to revert, but it's ok for some to. If there are no queueable rollbacks when the handler is called, it's fine for that call to be a revert.